### ISSUE TITLE
serialize stage?

### ISSUE THREAD

**Issue by hainm** [arose/ngl#308]
what do you think? 

(currently is for https://github.com/jupyter-widgets/ipywidgets/issues/1038#issue-200694339)

PS: I tried with JSON.stringify but getting `Converting circular structure to JSON`

**Comment by hainm** [comment]
May be

```python
NGL.serialize(stage)
stage = NGL.deserialize(dict)
```

**Comment by arose** [comment]
Yeah, definitely planned. Just doing JSON.stringify (resolving the circular structure issue) will result in huge objects. This needs more special handling of the data for structures, volumes, etc. The rest can then be a JSON object describing the state.

**Comment by arose** [comment]
related #97

**Comment by harryjubb** [comment]
Any progress in this area? Interested in this in the longer term
